### PR TITLE
Fix firewall creation by passing network

### PR DESCRIPTION
**Pull request by rtitle**
I manually deployed this to dev in order to test with AoU… it seems to work.

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/VPCHelper.scala [hunk 1]
```diff
@@ -16,40 +16,43 @@ class VPCHelper(config: VPCHelperConfig,
...
     for {
-      rule <- googleComputeService.getFirewallRule(googleProject, config.firewallRuleName)
-      _ <- rule.fold(
-        googleComputeService.addFirewallRule(googleProject, buildFirewall(googleProject, regionName, vpcConfig))
-      )(
-        _ => IO.unit
-      )
+      ruleOpt <- googleComputeService.getFirewallRule(googleProject, config.firewallRuleName)
+      _ <- if (ruleOpt.isEmpty) {
+        // Firewall rules don't work with a subnet, so we have to pass a network. This is
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/VPCHelper.scala:38]: The issue in the last PR is that firewall rules require a network, not a subnet… sigh

**Review by Qi77Qi — APPROVED** [review]

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/VPCHelper.scala (hunk 1 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/VPCHelper.scala:38]: so what was the initial bug before last PR?

Comment (rtitle) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/VPCHelper.scala:38]: I'm a bit confused how it used to work before we used `VPCHelper`/`GoogleComputeService`. Looking at old code, it seemed we passed the subnet to the firewall creation request in `HttpGoogleComputeDAO` (which is why I initially did that in my last PR). It's possible that the old code wasn't correct either and it was working by accident somehow.

Comment (kyuksel) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/VPCHelper.scala:38]: This is so tricky 😞

**Review by kyuksel — APPROVED** [review]
Thanks for fixing this. Given how tricky it is and how much trouble it's created, we should probably try to do more regression testing around it. I wonder if [Firewall Rules Logging](https://cloud.google.com/vpc/docs/firewall-rules-logging) may help too.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/VPCHelper.scala (hunk 1 above)
Comment (rtitle) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/VPCHelper.scala:38]: It really is.. before the post-mortem I'll try and better understand this change in behavior.

**Comment by rtitle** [comment]
> Given how tricky it is and how much trouble it's created, we should probably try to do more regression testing around it. I wonder if Firewall Rules Logging may help too.

Yeah, I would say this definitely highlighted a big testing gap. That link is interesting, I wonder if we could do more monitoring of notebooks traffic at the network level.

**Pull request merged**